applications/ThermalDEMApplication/custom_strategies/strategies/thermal_explicit_solver_strategy.cpp

```diff
@@ -73,6 +88,50 @@ namespace Kratos {
     KRATOS_CATCH("")
   }
 
+  //------------------------------------------------------------------------------------------------------------
+  void ThermalExplicitSolverStrategy::SetSolveFrequency(void) {
+    KRATOS_TRY
+    
+    // Parameters
+    ProcessInfo& r_process_info = GetModelPart().GetProcessInfo();
+    const double safety_factor  = 0.1;
+    const int    max_freq       = 10000;
+
+    // Get minimum diffusivity time between all particles
+    const int number_of_particles = (int)mListOfSphericParticles.size();
+    double min_diff_time = DBL_MAX;
+
+    #pragma omp parallel for
```

Comment:
I think now we are moving to parallel_utilities native in c++. It is better to use it since it supports exception handling in parallel (omp does not). see https://github.com/KratosMultiphysics/Kratos/wiki/Parallel-Utilities

Reply:
Moved